### PR TITLE
refactor: Replace `_.flatten` with `Array.prototype.flat`

### PR DESCRIPTION
**Pull request by pjmattingly**
Addresses: #7747

### CODE REVIEW

File: lib/plugins/aws/deploy/lib/check-for-changes.js [hunk 1]
```diff
@@ -255,6 +255,8 @@ module.exports = {
     const partition = account.partition;
 
     const functionNames = await this.serverless.service.getAllFunctions();
+
+    // #7747, no tests for this file, so did not remove _.flatten()
```
Comment (pgrzesik) [inline lib/plugins/aws/deploy/lib/check-for-changes.js:259]: How did you determine that there are no tests for this file? There are pretty new tests that cover this logic: https://github.com/serverless/serverless/blob/e7ad5aa268b7b0e10df2c6199e4fd93415de3c30/test/unit/lib/plugins/aws/deploy/lib/check-for-changes.test.js#L1146

**Review by pgrzesik — CHANGES_REQUESTED** [review]
Thanks @pjmattingly, I have one comment, please let me know what do you think

File: lib/plugins/aws/deploy/lib/check-for-changes.js (hunk 1 above)
Comment (pjmattingly) [reply lib/plugins/aws/deploy/lib/check-for-changes.js:259]: > pretty new tests

That's probably it. I see the tests on the latest sync. Let me see about changes to this file.

Comment (pjmattingly) [reply lib/plugins/aws/deploy/lib/check-for-changes.js]: I added the change in the latest, and testing looks good.

**Review by pgrzesik — APPROVED** [review]
Thank you @pjmattingly, it looks good 👍

**Pull request merged**